lib/compiler-llvm/src/config.rs

```diff
@@ -177,16 +192,47 @@ impl LLVM {
 
         let target_triple = self.target_triple(target);
         let llvm_target = InkwellTarget::from_triple(&target_triple).unwrap();
-        llvm_target
+        let llvm_target_machine = llvm_target
             .create_target_machine(
                 &target_triple,
-                "generic",
-                &llvm_cpu_features,
+                match triple.architecture {
+                    Architecture::Riscv64(_) => "generic-rv64",
+                    _ => "generic",
+                },
+                match triple.architecture {
+                    Architecture::Riscv64(_) => "+m,+a,+c,+d,+f",
+                    _ => &llvm_cpu_features,
+                },
                 self.opt_level,
                 self.reloc_mode(),
-                self.code_model(),
+                match triple.architecture {
+                    Architecture::Riscv64(_) => CodeModel::Medium,
+                    _ => self.code_model(),
+                },
             )
-            .unwrap()
+            .unwrap();
+
+        if let Architecture::Riscv64(_) = triple.architecture {
+            // TODO: totally non-portable way to change ABI
+            unsafe {
+                pub struct MyTargetMachine {
```

Comment:
Any particular reason to wrap this in a struct rather than just writing to the memory directly?

If sticking with the struct

* should have `#[repr(c)]` 
* a comment explaining what's going would be nice

Reply:
Added some more comment.
This must not have the `#[repr(c)]` decorator, as the point is to have the same memory representation

Reply:
Super hacky, but what works that works.